kmip/tests/unit/services/server/test_engine.py

```diff
@@ -4361,6 +4363,199 @@ def test_activate_not_allowed_by_policy(self):
+        reason_compromise = objects.RevocationReason(
+            code=enums.RevocationReasonCode.KEY_COMPROMISE)
+        date = primitives.DateTime(
+            tag=enums.Tags.COMPROMISE_OCCURRENCE_DATE, value=6)
+
+        # Test that reason UNSPECIFIED will put object into state
+        # DEACTIVATED
+        payload = revoke.RevokeRequestPayload(
+            unique_identifier=attributes.UniqueIdentifier(object_id),
+            revocation_reason=reason_unspecified,
+            compromise_date=date)
+
+        response_payload = e._process_revoke(payload)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        e._logger.info.assert_any_call(
+            "Processing operation: Revoke"
+        )
+        self.assertEqual(
+            str(object_id),
+            response_payload.unique_identifier.value
+        )
+
+        symmetric_key = e._data_session.query(
+            pie_objects.SymmetricKey
+        ).filter(
+            pie_objects.ManagedObject.unique_identifier == object_id
+        ).one()
+
+        self.assertEqual(enums.State.DEACTIVATED, symmetric_key.state)
+
+        # Test that reason KEY_COMPROMISE will put object into state
+        # COMPROMISED
+        payload = revoke.RevokeRequestPayload(
+            unique_identifier=attributes.UniqueIdentifier(object_id),
+            revocation_reason=reason_compromise,
+            compromise_date=date)
+
+        response_payload = e._process_revoke(payload)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        e._logger.info.assert_any_call(
+            "Processing operation: Revoke"
+        )
+        self.assertEqual(
+            str(object_id),
+            response_payload.unique_identifier.value
+        )
+
+        symmetric_key = e._data_session.query(
+            pie_objects.SymmetricKey
+        ).filter(
+            pie_objects.ManagedObject.unique_identifier == object_id
+        ).one()
+
+        self.assertEqual(enums.State.COMPROMISED, symmetric_key.state)
+
+        # Test that the ID placeholder can also be used to specify revocation.
+        e._id_placeholder = str(object_id)
+        payload = revoke.RevokeRequestPayload(
+            revocation_reason=reason_unspecified,
+            compromise_date=date)
+
+        response_payload = e._process_revoke(payload)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        e._logger.info.assert_any_call(
+            "Processing operation: Revoke"
+        )
+        self.assertEqual(
+            str(object_id),
+            response_payload.unique_identifier.value
+        )
+
+        symmetric_key = e._data_session.query(
+            pie_objects.SymmetricKey
+        ).filter(
+            pie_objects.ManagedObject.unique_identifier == object_id
+        ).one()
+
+        self.assertEqual(enums.State.DEACTIVATED, symmetric_key.state)
```

Comment:
The whole test looks good. Just update it to test for those specific state cases. Feel free to break this into smaller test functions too if you want.